src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -999,12 +999,10 @@ OverlayImpl::processValidatorList(
 
     // Convert the boost::string_view, returned by
     // boost::http::request::target(), into a std::string_view.
-    //    std::string_view key = [&req, &prefix]() {
-    //        std::string_view const key = req.target().substr(prefix.size());
-    //        return key;
-    //    }();
-
-    std::string_view key = req.target().substr(prefix.size());
+    std::string_view key = [&req, &prefix]() {
+        boost::string_view const key = req.target().substr(prefix.size());
+        return std::string_view(key.data(), key.length());
+    }();
```

Comment:
What was wrong with the old code? Specifically:

```cpp
std::string_view key = req.target().substr(prefix.size());
```

Reply:
the constructor of `std::string_view` reads a `char*` pointer until null-termination. I observed some garbage values at the end of `key` when the second length parameter wasn't specified

Reply:
Check out this thread: https://github.com/XRPLF/rippled/pull/4509#discussion_r1176760881

Reply:
oh you mean: explicitly specify the type `std::string_view` instead of `auto` ? that could work

Reply:
What does `req.target().substr(...)` return? A `boost::string_view`? Is it implicitly convertible to `char*` or something? Where does the `char*` come in?

Reply:
Yes, it returns a `boost::string_view`. I don't remember the internal mechanics exactly, I remember coming across the garbage values. So I decided to use the length parameter.

Now, I have used your suggestion as-is. I have updated my conan profile to use the `BOOST_BEAST_STD_STRING` macro. Ever since, the inter-conversion from `boost::string_view` to `std::string_view` has occured without a hitch.

fixed in [88412ea](https://github.com/XRPLF/rippled/pull/4509/commits/88412eaa6e86782489f073d724fc0acf9d6e6321)